runtime/src/platform_utils.cpp

```diff
@@ -0,0 +1,49 @@
+#include "platform_utils.hpp"
+
+#if __linux__
+#include <sched.h>
+
+int my_sched_getcpu() {
+    return ::sched_getcpu();
+}
+
+#elif __APPLE__
+#include <sys/sysctl.h>
+#include <pthread.h>
+
+int my_sched_getcpu() {
+    int cpu = -1;
+    size_t len = sizeof(cpu);
+
+    if (sysctlbyname("hw.cpulocation", &cpu, &len, NULL, 0) == -1) {
+        return -1;  // Handle error
+    }
+    return cpu;
+}
+
+int sched_getaffinity(pid_t pid, size_t cpusetsize, cpu_set_t *mask) {
+    (void)pid;
+    (void)cpusetsize;
+    CPU_ZERO(mask);
+    CPU_SET(my_sched_getcpu(), mask);
+    return 0;
+}
+
+int sched_setaffinity(pid_t pid, size_t cpusetsize, const cpu_set_t *mask) {
+    (void)pid;
+    (void)cpusetsize;
+    (void)mask;
+    return 0;
+}
+
+#endif
+
+int get_current_cpu() {
```

Comment:
What's the difference between `get_current_cpu()` and `my_sched_getcpu()`? I see them being used interchangeably

Reply:
you are right, there should be only one, i will update the header. Thanks.